test/extended/oauth/well_known.go

```diff
@@ -0,0 +1,42 @@
+package oauth
+
+import (
+	"encoding/json"
+	"net/http"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/origin/pkg/oauth/util"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Suite:openshift/oauth] The OAuth server well-known endpoint", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLI("oauth-well-known", exutil.KubeConfigPath())
+
+	g.It("should be reachable", func() {
+		metadataJSON, err := oc.Run("get").Args("--raw", "/.well-known/oauth-authorization-server").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		metadata := &util.OauthAuthorizationServerMetadata{}
+		err = json.Unmarshal([]byte(metadataJSON), metadata)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(metadata.AuthorizationEndpoint).Should(o.ContainSubstring("authorize"))
```

Comment:
Get the hostname from the route in `openshift-authentication` and check that the endpoint is correctly configured

Reply:
ok, got it, done